From a006d7a6cfacc88772951d148496284da3c95cf6 Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Sat, 30 Apr 2016 13:05:43 -0700
Subject: [PATCH 27/45] Fix pager mini-index drawing when $pager_index_lines =
 1 and $status_on_top.

When $pager_index_lines is 1, there is room for the index status line,
but not for any actual index entries.  The index redraw part was
always calling menu_redraw_current() which doesn't actually check if
there is room for the current row.

This bug only shows up with $status_on_top, due to the order things
are redrawn.
---
 pager.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pager.c b/pager.c
index 822c31a..f258aae 100644
--- a/pager.c
+++ b/pager.c
@@ -1880,7 +1880,8 @@ mutt_pager (const char *banner, const char *fname, int flags, pager_t *extra)
     {
       /* redraw the pager_index indicator, because the
        * flags for this message might have changed. */
-      menu_redraw_current (index);
+      if (index_window->rows > 0)
+        menu_redraw_current (index);
 #ifdef USE_SIDEBAR
       mutt_sb_draw();
 #endif
-- 
2.8.2

